common/config/config.go

```diff
@@ -321,6 +325,8 @@ type (
 		RPCName string `yaml:"rpcName"`
 		// Address indicate the remote service address(Host:Port). Host can be DNS name.
 		RPCAddress string `yaml:"rpcAddress"`
+		// Private Key Path
+		PrivateKey string `yaml:"privateKey"`
```

Comment:
Let's add a new `AuthorizationProvider` struct here
AuthorizationProvider struct{
  Enable bool
  Type string  // only support OAuthAuthorization
  PrivateKey string
}

Also we can remove the private key from JwtCredentials and only use this one: https://github.com/uber/cadence/blob/a58b8b9670417ccbf2cd021882e750f95d6fc508/common/config/config.go#L104